### PR TITLE
Add SYN_LOG_DATEFMT for logging (SYN-6217)

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: docs/synapse/devopsguide.rst [hunk 1]
```diff
@@ -292,6 +292,21 @@ log message pretty-printed log message::
       "user": "3189065f95d3ab0a6904e604260c0be2"
     }
 
+Custom date formatting strings can also be provided by setting the ``SYN_LOG_DATEFMT`` string. These is expected to be
```
Comment (Cisphyx) [inline docs/synapse/devopsguide.rst:295]: "These is" should be "This is"

File: docs/synapse/devopsguide.rst [hunk 2]
```diff
@@ -292,6 +292,21 @@ log message pretty-printed log message::
...
+    SYN_LOG_DATEFMT="%d%m%Y %H:%M:%S"
+
+produces teh following output::
+
+    28062021 15:48:01 [INFO] log level set to DEBUG [common.py:setlogging:MainThread:MainProcess]
+
+This will also be used to format the ``time`` key used for structured logging.
+
+.. warning::
+    Milliseconds are not available when using the date formatting option. This will result in a lose of precision for
```
Comment (Cisphyx) [inline docs/synapse/devopsguide.rst:307]: should be "loss" instead of "lose"

File: synapse/common.py [hunk 3]
```diff
@@ -793,7 +794,7 @@ def normLogLevel(valu):
             return normLogLevel(valu)
     raise s_exc.BadArg(mesg=f'Unknown log level type: {type(valu)} {valu}', valu=valu)
 
-def setlogging(mlogger, defval=None, structlog=None, log_setup=True):
```
Comment (Cisphyx) [inline synapse/common.py:796]: the docstring for this function could use an update

File: docs/synapse/devopsguide.rst [hunk 4]
```diff
@@ -292,6 +292,21 @@ log message pretty-printed log message::
       "user": "3189065f95d3ab0a6904e604260c0be2"
     }
 
+Custom date formatting strings can also be provided by setting the ``SYN_LOG_DATEFMT`` string. This is expected to be a
+strftime_ format string. The following shows an example of setting this value::
+
+    SYN_LOG_DATEFMT="%d%m%Y %H:%M:%S"
```
Comment (invisig0th) [inline docs/synapse/devopsguide.rst:298]: Lets use `SYN_LOG_DATEFORMAT` here ( no need to rename internal vars or anything ). Clarity is worth more than brevity in this instance :)

**Review by invisig0th — CHANGES_REQUESTED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**